openfe/setup/box_representation.py

```diff
@@ -0,0 +1,104 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+import numpy as np
+
+
+def is_valid_box(box: np.ndarray) -> bool:
+    """Check that a box is valid
+
+    Current rules:
+    - the y and z component of the x vector must be zero
+    - the z component of the y vector must be zero
+
+    Parameters
+    ----------
+    box : np.ndarray, shape(3,3)
+       the simulation cell
+
+    Returns
+    -------
+    valid : bool
+    """
+    if box.shape != (3, 3):
+        return False
+    lx, xy, xz = box[0]
+    yx, ly, yz = box[1]
+    zx, zy, lz = box[2]
+
+    if (xy != 0.0) or (xz != 0.0) or (yz != 0.0):
+        return False
+    # TODO: Max tilt factors, are these common across all MD engines?
+    return True
+
+
+class BoxRepresentation:
```

Comment:
Beyond our other conversations re: maybe not having this defined in all cases, do we need to account for a `None` version of this class? (thinking of say a gas phase calculation)

Reply:
So we'd previously agreed that this Box information is optional, and only really necessary if the box size is important to the system (e.g. Membrane), but could be invented by the method if required (e.g. adding padding around the "interesting" parts).  Letting the method define the box ensured compatibility between different nodes.

I think `Vacuum` as a Box is good because it's saying the box isn't implicit here (as above) but instead there really is no box.  The alternative would be to make the above case `AutoBox` to signal to the method that it can invent a box, and the lack of a box meant Vacuum, but I think that's a more clunky option.

Reply:
Yeah `Vacuum` sounds like a good idea.

Reply:
In this case, is `Vacuum` an instance, a class, or just the string?